searchcore/src/vespa/searchcore/proton/server/document_scan_iterator.cpp

```diff
@@ -23,12 +23,12 @@ DocumentScanIterator::valid() const
 }
 
 DocumentMetaData
-DocumentScanIterator::next(uint32_t compactLidLimit, uint32_t maxDocsToScan, bool retry)
+DocumentScanIterator::next(uint32_t compactLidLimit, bool retry)
 {
     if (!retry) {
         --_lastLid;
     }
-    for (uint32_t i(0); (_lastLid > compactLidLimit) && (i < maxDocsToScan); ++i, --_lastLid) {
+    for (uint32_t i(0); _lastLid > compactLidLimit; ++i, --_lastLid) {
```

Comment:
We can remove the _i_ variable as well now.

Reply:
Fixed in https://github.com/vespa-engine/vespa/pull/16135